mlflow/server/js/src/components/MetricsPlotPanel.js

```diff
@@ -220,12 +221,19 @@ export class MetricsPlotPanel extends React.Component {
   handleLayoutChange = (newLayout) => {
     // Unfortunately, we need to parse out the x & y axis range changes from the onLayout event...
     // see https://plot.ly/javascript/plotlyjs-events/#update-data
-    const newXRange0 = newLayout["xaxis.range[0]"];
-    const newXRange1 = newLayout["xaxis.range[1]"];
-    const newYRange0 = newLayout["yaxis.range[0]"];
-    const newYRange1 = newLayout["yaxis.range[1]"];
+    const {
+      "xaxis.range[0]": newXRange0,
+      "xaxis.range[1]": newXRange1,
+      "yaxis.range[0]": newYRange0,
+      "yaxis.range[1]": newYRange1,
+      "xaxis.autorange": xAxisAutorange,
+      "yaxis.autorange": yAxisAutorange,
+      ...restFields
+    } = newLayout;
+
     let mergedLayout = {
       ...this.state.layout,
+      ...restFields,
     };
```

Comment:
@smurching 
I hope this is what you meant in https://github.com/mlflow/mlflow/pull/2459#issuecomment-589999741

Reply:
@harupy yeah looks good!